api/src/main/java/org/eclipse/microprofile/metrics/annotation/ConcurrentGauge.java

```diff
@@ -0,0 +1,128 @@
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.enterprise.util.Nonbinding;
+import javax.interceptor.InterceptorBinding;
+
+/**
+ * An annotation for marking a method, constructor, or class as concurrent gauged.
+ * The metric will be registered in the application MetricRegistry.
+ *
+ * A concurrent gauge has the following semantics:
+ * <ul>
+ *   <li>Upon entering the marked item, the value is increased.</li>
+ *   <li>Upon exiting the marked item, the value is decreased.</li>
+ * </ul>
+ *
+ * <p>
+ * Given a method annotated with {@literal @}ConcurrentGauge like this:
+ * </p>
+ * <pre><code>
+ *     {@literal @}ConcurrentGauge(name = "fancyName")
+ *     public String fancyName(String name) {
+ *         return "Sir Captain " + name;
+ *     }
+ * </code></pre>
+ * A concurrent gauge with the fully qualified class name + {@code fancyName} will be created and each time the
+ * {@code #fancyName(String)} method is invoked, the counter will be marked.
+ * Similarly, the same applies for a constructor annotated with counted.
+ *
+ * <p>
+ * Given a class annotated with {@literal @}ConcurrentGauge like this:
+ * </p>
+ * <pre><code>
+ *     {@literal @}ConcurrentGauge
+ *     public class CGaugedBean {
+ *         public void cGaugedMethod1() {}
+ *         public void cGaugedMethod2() {}
+ *     }
+ * </code></pre>
+ * A counter for the defining class will be created for each of the constructors/methods.
+ * Each time the constructor/method is invoked, the respective counter will be marked.
+ *
+ * @since 2.0
+ */
+@Inherited
+@Documented
+@InterceptorBinding
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.TYPE, ElementType.CONSTRUCTOR, ElementType.METHOD })
+public @interface ConcurrentGauge {
+
+    /**
+     * @return The name of the concurrent gauge.
+     */
+    @Nonbinding
+    String name() default "";
+
+    /**
+     * @return The tags of the concurrent gauge. Each {@code String} tag must be in the form of 'key=value'. If the input is empty or does
+     * not contain a '=' sign, the entry is ignored.
+     *
+     * @see org.eclipse.microprofile.metrics.Metadata
+     */
+    @Nonbinding
+    String[] tags() default {};
+
+    /**
+     * @return If {@code true}, use the given name as an absolute name. If {@code false} (default), use the given name
+     * relative to the annotated class. When annotating a class, this must be {@code false}.
+     */
+    @Nonbinding
+    boolean absolute() default false;
+
+
+    /**
+     * @return The display name of the concurrent gauge.
+     *
+     * @see org.eclipse.microprofile.metrics.Metadata
+     */
+    @Nonbinding
+    String displayName() default "";
+
+    /**
+     * @return The description of the concurrent gauge.
+     *
+     * @see org.eclipse.microprofile.metrics.Metadata
+     */
+    @Nonbinding
+    String description() default "";
+
+
```

Comment:
Should there be a unit's parameter to match up with the other annotations?

Reply:
Theoretically yes to be consistent with other metrics. In reality the unit would always be NONE.